### PR TITLE
Add support for tsconfig extends

### LINKED ISSUE #28: Error! Unexpected token ] in "tsconfig.json" at position 666
My tsconfig.json is : 

    {
      "compilerOptions": {
        "target": "esnext",
        "module": "commonjs",
        "lib": ["dom", "es2015", "esnext"],
        "outDir": "dist",
        "jsx": "react",
        "baseUrl": "./src",
        "paths": {
          "@src/*": ["src/*"],
          "@sections/*": ["src/app/sections/*"],
          "@app/*": ["src/app/*"]
        },
        "strict": true,
        "sourceMap": true,
        "skipLibCheck": true,
        "noImplicitAny": false,
        "noImplicitThis": false,
        "moduleResolution": "node",
        "esModuleInterop": true,
        "allowSyntheticDefaultImports": true,
        "resolveJsonModule": true,
        "allowJs": true
      },
      "include": ["src/**/*"],
      "exclude": [
        "src/index.js",
        "dist",
      ]
    }

Executing `alias-hq CLI` : 

    ./node_modules/.bin/alias-hq 

      == Alias HQ ==

      Error! Unexpected token ] in "tsconfig.json" at position 666

      Edit the file to fix this, then try again

Why? How to solve the problem?

    typescript : v.4.1.5

### LINKED ISSUE #30: Accessing paths from a `tsconfig.json` that extends another `tsconfig.base.json` with the alias paths
👋 Hey, great library! I've got this working for the most part in a monorepo structure, but I'm having trouble with `hq.get('any')` returning `{}` when used inside the `/package/*` directory roots. This is because the `tsconfig.json` inside each package extends the `tsconfig.base.json` at the monorepo root where I have my `paths` configured. Because the `tsconfig.json` doesn't explicitly contain the `paths`, the result is `{}` and thus I'm having issues plugging in aliases in some package-specific processes like Jest.

For the most part everything else works fine, and I've got a nice clean, centralised alias structure which all runs and builds fine - just trying to figure out this last piece.

Here's the structure:

```
monorepo/
    ...
    tsconfig.base.json (1)
    packages/
        client/
            ...
            tsconfig.json (2)
        components/
            ...
            tsconfig.json (3)
```

In the monorepo root I've got a couple processes that consume the aliases set in `tsconfig.base.json` (1), e.g. a shared Storybook across all packages. In `tsconfig.base.json` I have my alias paths set up like so:
```
{
    "compilerOptions": {
        ...
        "rootDir": "./",
        "baseUrl": "./packages",
        "paths": {
            "@client/*": ["../packages/client/*"],
            "@components/*": ["../packages/components/*"]
        }
    }
}
```
and in my packages, e.g. `client`, I have a `tsconfig.json` (2) that extends the root `tsconfig.base.json` (3):
```
{
    "extends": "../../tsconfig.base.json",
    "compilerOptions": {
        ...
        "baseUrl": "../",
     }
}
```
This works in that my code compiles, runs, builds etc and resolves the aliases, but as I've said, if I try and use `hq.get('any')` ('webpack', 'jest', w/e) in any of the package roots it returns empty. It returns the paths just fine in the root. Sorry for the repetitive explanation, I hope I've made my issue clear.

Thanks again!

### PR DESCRIPTION
**Pull request by IanVS**
This uses the native typescript utilities for reading and parsing a config file, which means that comments and trailing commas are supported, as well as extending from another config.

I needed to break out a helper function to log messages, since the parsing is slightly different between package.json and config files now, but they can share the same error handling, I think.

I also created a few basic tests for loading config files with comments and commas, as well as extending from other configs.  Let me know if you'd like to see more.

Closes #28 
Closes #30

### CODE REVIEW

**Comment by davestewart** [comment]
Looks good!

It's late here so I'll take a look tomorrow.

Thanks :)

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const Path = require('path')
 const Fs = require('fs')
 const { resolve } = require('./utils')
+const ts = require('typescript')
```
Comment (davestewart) [inline src/index.js:4]: Can you name this `Ts` (cap T) and move to top of file

File: src/index.js [hunk 2]
```diff
@@ -94,29 +123,12 @@ function makeConfig () {
...
-        console.warn(message)
-        process.exit(0)
-      }
-
-      // API
-      console.warn(`\n  [Alias HQ]\n${message}\n`.red)
-      throw(err)
-    }
+  const text = Fs.readFileSync(path, 'utf8').toString()
+  const {config, error} = ts.parseConfigFileTextToJson(path, text)
```
Comment (davestewart) [inline src/index.js:127]: I haven't added any linting to this project yet (will do that). I use Standard JS with Stroustrop brackets, so:

- spaces round deconstruction braces please
- else on newline

File: src/index.js [hunk 3]
```diff
@@ -142,14 +154,18 @@ function loadSettings () {
  */
 function loadConfig (path) {
   // load
-  const json = loadJson(path)
+  const text = Fs.readFileSync(path, 'utf8').toString()
+  const {config: json, error} = ts.parseConfigFileTextToJson(path, text)
```
Comment (davestewart) [inline src/index.js:158]: spaces round brackets please

File: src/index.js [hunk 4]
```diff
@@ -142,14 +154,18 @@ function loadSettings () {
...
 
-  // config
-  const compilerOptions = json && json.compilerOptions
+  // extract config
+  const compilerOptions = json && ts.parseJsonConfigFileContent(json, ts.sys, Path.dirname(path)).options
   if (compilerOptions) {
-    // config
-    config.rootUrl = Path.dirname(path)
-    config.baseUrl = compilerOptions.baseUrl || ''
+    config.rootUrl = compilerOptions.pathsBasePath || Path.dirname(path);
```
Comment (davestewart) [inline src/index.js:166]: No terminating semis

File: tests/specs/load.spec.js [hunk 5]
```diff
@@ -38,6 +38,21 @@ describe('calling load and passing', function () {
...
+      const getConfig = () => hq.load('demo/tsconfig-fancy.json')
+      expect(getConfig).not.toThrowError();
+    });
+  })
+
+  describe('config extending another config', () => {
+    it('should extract paths from the base config', () => {
+      const received = hq.load('demo/tsconfig-fancy.json')
+      const expected = config.ts
+      expect(received.config.paths).toEqual(expected);
```
Comment (davestewart) [inline tests/specs/load.spec.js:52]: No semis, thanks

**Review by davestewart — CHANGES_REQUESTED** [review]
All good! Just some code style changes, thanks :)

**Comment by davestewart** [comment]
OK, linting now added so you'll want to merge and run `npm run lint`.

**Comment by IanVS** [comment]
Updated to address linting issues

**Review by davestewart — APPROVED** [review]

**Comment by davestewart** [comment]
No idea what these errors are about. Any ideas?

**Comment by davestewart** [comment]
Just reviewing this as I need it today!

This lib came up in my Google search:

- https://github.com/TypeStrong/tsconfig

I'll probably look to implement it...

**Pull request merged**